railgun/internal/diagnostics/server.go

```diff
@@ -0,0 +1,54 @@
+package diagnostics
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/kong/kubernetes-ingress-controller/railgun/internal/mgrutils"
+	"github.com/kong/kubernetes-ingress-controller/railgun/pkg/config"
+	"github.com/sirupsen/logrus"
+)
+
+type diagnosticsServer struct {
+	logger logrus.FieldLogger
+	mux    *http.ServeMux
+}
+
+func NewDiagnosticsServer(enableProfiling bool, log logrus.FieldLogger) *diagnosticsServer {
```

Comment:
To simplify the implementation, we can remove `NewDiagnosticsServer` altogether:
- remove the `mux` field from the struct and make it a local variable in `Start()` and initialize it there
- remove `NewDiagnosticsServer` and, with the comment above (which makes the type exported) addressed, export the `Logger` field of `Server`

Therefore the initialization becomes:
```go
s := diagnostics.Server{Logger: util.MakeLogger(c.LogLevel, c.LogFormat)}
if err := s.Listen(ctx); err != nil { ... } // renamed in another comment
```